### PR TITLE
Mbaas/fix expiration time

### PR DESCRIPTION
**Pull request by tjyung**
* [x] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you add unit tests?
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?
* [x] Did you check UI tests on the sample app? They are not executed on CI.

## Description

Fixed the expiration time check and updated write conditions when we call cosmos list.

### CODE REVIEW

File: sdk/appcenter-data/src/main/java/com/microsoft/appcenter/data/models/LocalDocument.java [hunk 1]
```diff
@@ -129,7 +131,10 @@ public long getOperationTime() {
      * @return whether the document is expired
      */
     public boolean isExpired() {
-        return getExpirationTime() <= System.currentTimeMillis();
+        if (mExpirationTime == TimeToLive.INFINITE){
```
Comment (guperrot) [inline sdk/appcenter-data/src/main/java/com/microsoft/appcenter/data/models/LocalDocument.java]: code is not formatted

File: sdk/appcenter-data/src/test/java/com/microsoft/appcenter/data/UtilsTest.java [hunk 2]
```diff
@@ -78,6 +79,11 @@ public void getETag() {
         assertNull(Utils.getETag("{a:1}"));
     }
 
+    @Test
+    public void localDocumentExpired() {
+        LocalDocument localDocument = new LocalDocument(DefaultPartitions.APP_DOCUMENTS, null, "user", "test", "test", TimeToLive.INFINITE, 0, 0);
+        assertFalse(localDocument.isExpired());
+    }
```
Comment (guperrot) [inline sdk/appcenter-data/src/test/java/com/microsoft/appcenter/data/UtilsTest.java]:
```suggestion
    }

```

File: sdk/appcenter-data/src/main/java/com/microsoft/appcenter/data/models/LocalDocument.java (hunk 1 above)
Comment (guperrot) [reply sdk/appcenter-data/src/main/java/com/microsoft/appcenter/data/models/LocalDocument.java]:
```suggestion
        if (mExpirationTime == TimeToLive.INFINITE) {
```

File: sdk/appcenter-data/src/main/java/com/microsoft/appcenter/data/Data.java [hunk 3]
```diff
@@ -627,7 +627,7 @@ public void onCallSucceeded(String payload, Map<String, String> headers) {
                         if (items != null) {
                             for (DocumentWrapper<T> document : items) {
                                 if (document.getError() == null) {
-                                    mLocalDocumentStorage.writeOnline(tableName, document, new WriteOptions());
+                                    mLocalDocumentStorage.writeOnline(tableName, document, new WriteOptions(readOptions.getDeviceTimeToLive()));
```
Comment (guperrot) [inline sdk/appcenter-data/src/main/java/com/microsoft/appcenter/data/Data.java]: There is no new unit test for this.

File: sdk/appcenter-data/src/main/java/com/microsoft/appcenter/data/models/LocalDocument.java (hunk 1 above)
Comment (guperrot) [reply sdk/appcenter-data/src/main/java/com/microsoft/appcenter/data/models/LocalDocument.java]: Also this bug on infinite probably exists in 2.0.0, thus it should be in changelog.md

Comment (YiMysty) [reply sdk/appcenter-data/src/main/java/com/microsoft/appcenter/data/models/LocalDocument.java]: sure :)

File: CHANGELOG.md [hunk 4]
```diff
@@ -11,6 +11,7 @@
 ### App Center Data
 
 * **[Feature]** Change the default time-to-live (TTL) from 1 day to infinite (never expire).
+* **[Fix]** Fix `isExpired` method in `LocalDocument` for incorrect handling `INFINITE` ttl situation.
```
Comment (guperrot) [inline CHANGELOG.md]:
```suggestion
* **[Fix]** Fix `isExpired` method in `LocalDocument` for incorrect handling of the `TimeToLive.INFINITE` value. 
```

File: sdk/appcenter-data/src/test/java/com/microsoft/appcenter/data/DataReadTest.java [hunk 5]
```diff
@@ -149,6 +149,7 @@ public void readCosmosDbCallEncodeDocumentId() throws JSONException, Unsupported
                 notNull(ServiceCallback.class));
     }
 
+
```
Comment (guperrot) [inline sdk/appcenter-data/src/test/java/com/microsoft/appcenter/data/DataReadTest.java]:
```suggestion
```

**Pull request merged**